ecl/hthor/hthor.cpp

```diff
@@ -10363,6 +10364,681 @@ void CHThorExternalActivity::stop()
+            gatherInfo(fdesc);
+            IDistributedFile *dFile = ldFile->queryDistributedFile();
+            if (dFile)  //only makes sense for distributed (non local) files
+            {
+                dfsParts.setown(dFile->getIterator());
+                IDistributedSuperFile *super = dFile->querySuperFile();
+                if (super)
+                {
+                    unsigned numsubs = super->numSubFiles(true);
+                    unsigned s=0;
+                    for (; s<numsubs; s++)
+                    {
+                        IDistributedFile &subfile = super->querySubFile(s, true);
+                        subfiles.append(*extractFileInformation(&subfile));
+                    }
+                    assertex(fdesc);
+                    superfile.set(fdesc->querySuperFileDescriptor());
+                }
+                else
+                    subfiles.append(*extractFileInformation(dFile));
+
+                if((helper.getFlags() & (TDXtemporary | TDXjobtemp)) == 0)
+                    agent.logFileAccess(dFile, "HThor", "READ");
+            }
+            else
+                subfiles.append(*extractFileInformation(nullptr));
+        }
+        else
+            subfiles.append(*extractFileInformation(nullptr));
+
+        if (!ldFile)
+        {
+            StringBuffer buff;
+            buff.appendf("Input file '%s' was missing but declared optional", mangledHelperFileName.str());
+            WARNLOG("%s", buff.str());
+            agent.addWuException(buff.str(), WRN_SkipMissingOptFile, SeverityInformation, "hthor");
+        }
+    }
+
+    assertex(subfiles.ordinality() != 0);
+}
+
+void CHThorNewDiskReadBaseActivity::gatherInfo(IFileDescriptor * fileDesc)
+{
+    if (fileDesc)
+    {
+        if (!agent.queryResolveFilesLocally())
+        {
+            grouped = fileDesc->isGrouped();
+            if (grouped != ((helper.getFlags() & TDXgrouped) != 0))
+            {
+                StringBuffer msg;
+                msg.append("DFS and code generated group info. differs: DFS(").append(grouped ? "grouped" : "ungrouped").append("), CodeGen(").append(grouped ? "ungrouped" : "grouped").append("), using DFS info");
+                WARNLOG("%s", msg.str());
+                agent.addWuException(msg.str(), WRN_MismatchGroupInfo, SeverityError, "hthor");
+            }
+        }
+        else
+            grouped = ((helper.getFlags() & TDXgrouped) != 0);
+    }
+    else
+    {
+        grouped = ((helper.getFlags() & TDXgrouped) != 0);
+    }
+}
+
+CHThorNewDiskReadBaseActivity::InputFileInfo * CHThorNewDiskReadBaseActivity::extractFileInformation(IDistributedFile * distributedFile)
+{
+    Owned<IPropertyTree> meta = createPTree();
+    unsigned actualCrc = helper.getDiskFormatCrc();
+    Linked<IOutputMetaData> actualDiskMeta = expectedDiskMeta;
+    bool compressed = false;
+    bool blockcompressed = false;
+
+    if (distributedFile)
+    {
+        //MORE:
```

Comment:
Not clear what this comment is telling me... is it still valid (and if so is there a JIRA to cover the additional work that is needed?)

Reply:
This probably does need addressing.  I think it is where it is reading a binary file that was sprayed as csv - don't use the structure it was sprayed as (see lines 8080 in original activity).